lib/sport_ngin_aws_auditor/scripts/audit.rb

```diff
@@ -56,60 +56,79 @@ def self.gather_data(class_type, tag_name)
       end
 
       def self.print_data(slack, environment, data, class_type)
+        updated_data = modify_data(data)
+
         if slack
-          print_to_slack(data, class_type, environment)
+          print_to_slack(updated_data, class_type, environment)
         elsif options[:reserved] || options[:instances]
           puts header(class_type)
-          data.each{ |key, value| say "<%= color('#{key}: #{value}', :white) %>" }
+          updated_data.each{ |key, value| say "<%= color('#{key}: #{value}', :white) %>" }
         else
           puts header(class_type)
-          data.each{ |key, value| colorize(key, value) }
+          updated_data.each{ |key, value| colorize(key, value) }
+        end
+      end
+
+      def self.modify_data(data)
+        updated_data = []
+        data.each do |key, value|
+          if key.include?(" with tag")
+            k, v = modify_tag_prints(key, value)
+            updated_data.push([k, v])
+          else
+            if value > 0
+              value = '+' << value.to_s
+            end
+            updated_data.push([key, value.to_s])
+          end
         end
+
+        return updated_data.sort_by { |key, value| [value, key] }
       end
 
       def self.colorize(key, value)
-        if key.include?(" with tag")
-          key, value = modify_tag_prints(key, value)
+        if value.include?("*")
           say "<%= color('#{key}: #{value}', :blue) %>"
-        elsif value < 0
+        elsif value.include?("-")
```

Comment:
This string matching is a bit hokey.  If the data is getting this complex, it should really be modeled as more structured classes for easier readability and testability.